docs/guide/src/dev/protobuf.md

```diff
@@ -0,0 +1,38 @@
+# Maintaining protobuf specs
+
+The Penumbra project dynamically generates code for interfacing
+with [gRPC]. The following locations within the repository
+are relevant:
+
+  * `proto/proto/penumbra/**/*.proto`, the developer-authored spec files
+  * `proto/src/gen/*.rs`, the generated output files
+  * `tools/proto-compiler/`, the build logic for the generation
```

Comment:
In this section it might be helpful to link to https://buf.build/penumbra-zone/penumbra and mention that we auto-publish our protobuf schemas with Buf, and rely on Buf for generating Go and Typescript packages, but generate the Rust code with our own tooling.

The Buf pages render fairly nice docs, so that's worth calling out.